File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.stream.*;
import java.util.*;
class Solution {
    public int solution(int[] citations) {
        
        Arrays.sort(citations);
        
        int count = 0;
        for(int i = citations.length - 1; citations[i] > count; i-- ){
            count++;
            if(i == 0) break;
        }
        
        return count;
    }
}

File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    # citations = [10,9,8]
    
    # citations = [9, 7, 6, 2, 1]
    answer = 0
    value = 0
    citations.sort(reverse=True)
    for i,v in enumerate(citations):
        value = max(value, min(i + 1, v))
            
    return value
    